tests/robustness/validate/watch.go

```diff
@@ -59,8 +59,8 @@ func validateBookmarkable(t *testing.T, report report.ClientReport) {
 }
 
 func validateOrdered(t *testing.T, report report.ClientReport) {
-	var lastEventRevision int64 = 1
 	for _, op := range report.Watch {
+		var lastEventRevision int64 = 1
```

Comment:
Not sure why change this. Aren't watch responses globally ordered for each client? In your test case `TestValidateWatch`, there is no clientId; do you intentionally verify that different clients may request an older revision?

Reply:
True, we expect client to get increasing revisions due to either fact that watch doesn't break or that users will usually want to reestablish watches on the following revision. However, from etcd perspective, those are independent watch request each providing its own revision to start watching from. It's not invalid that single client can start watching from rev 200, and after that decide to establish new watch from rev 100. 

As in case https://github.com/etcd-io/etcd/issues/16693, for some unknown reason etcd went back from revision 301 to 192 in the KV store. So from clients perspective it behaved correctly, after watch was broken it established the new watch on revision 192, even though it has previously seen revision 301. 

Goal of this issue is to remove assumption about sensible client behavior (not going back on watch), and just validate the watch responses. This should increase readability of robustness test reports as client misbehavior caused by etcd linearizability issue will no longer also report invalid watch.

Reply:
> etcd went back from revision 301 to 192 in the KV store. So from clients perspective it behaved correctly, after watch was broken it established the new watch on revision 192, even though it has previously seen revision 301

It's true. But in your test case, there is no watch establishment, so the revision shouldn't go back? 

The test change is OK. But I'd suggest you to have a deep dive to figure out why the revision go back. Let me know if you need my assistance or I misunderstood anything.

Reply:
https://github.com/etcd-io/etcd/blob/6a96ab7c09b4173406c8105c28d72d029ab50d4d/tests/robustness/traffic/client.go#L230

Reply:
> It's true. But in your test case, there is no watch establishment, so the revision shouldn't go back?

There is, but only in Kubernetes traffic. It runs a ListWatch loop with 100 ms timeout.

Look consists of Read and Watch
https://github.com/etcd-io/etcd/blob/6a96ab7c09b4173406c8105c28d72d029ab50d4d/tests/robustness/traffic/kubernetes.go#L67-L82

And watch breaks every 100ms to simulate client loosing connection
https://github.com/etcd-io/etcd/blob/6a96ab7c09b4173406c8105c28d72d029ab50d4d/tests/robustness/traffic/kubernetes.go#L190-L197